### PR TITLE
Add routes documentation

### PR DESCRIPTION
**Pull request by Ovyerus**
Deals with #1

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+### **Parameters**
+ - `poem` (string) (required) - Text to generate an image for.
+ - `font` (string) (optional) - Font to use for the poem. Will also pick a corresponding background at times. If the font is unsupported, it will fallback to a default font.
+
+Parameters can either be sent as a query string (`?poem=foobar`), or as a JSON body, with the latter taking precedence.
+
+### **Responses**
+#### **Content-Type: application/json**
+A client-side error was made, and the server was unable to generate an image as a result of this.
```
Comment (sr229) [inline API.md]: be a little bit more verbose who what went wrong in the case

File: API.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Sayori API Routes Documentation
+
+## `/generate`
+### **Supported Methods:** `GET`, `POST`
+
+### **Parameters**
+ - `poem` (string) (required) - Text to generate an image for.
+ - `font` (string) (optional) - Font to use for the poem. Will also pick a corresponding background at times. If the font is unsupported, it will fallback to a default font.
```
Comment (sr229) [inline API.md]: font should only accept specific arguments

File: API.md (hunk 1 above)
Comment (Ovyerus) [reply API.md]: The different errors say what happened.

File: main.py [hunk 3]
```diff
@@ -83,18 +85,20 @@ def break_text(text, font, max_width):
         return web.Response(status=400, text='{"error": "Missing required field: `poem`.", "code": 1}', content_type='application/json')
 
     if type(body['poem']) is not str:
-        return web.Response(status=400, text='{"error": "Field `poem` is not a string.", "code": 2}', content_type='application/json')
+        return web.Response(status=400, text='{"error": "Field `poem` is not a string.", "code": 2}',
```
Comment (sr229) [inline main.py]: doesn't look documented

File: main.py [hunk 4]
```diff
@@ -83,18 +85,20 @@ def break_text(text, font, max_width):
...
+                            content_type='application/json')
 
     if not body['poem']:
         return web.Response(status=400, text='{"error": "Field `poem` is empty.", "code": 3}', content_type='application/json')
 
-    poem = body['poem'].replace('\r', '').replace('\n', '\u2426')
+    if 'font' in body and body['font'] not in FONTS:
+        return web.Response(status=400,
+                            text='{{"error": "Unsupported font. Supported fonts are: \\"{}\\"", "code": 4}}'.format('\\", \\"'.join(FONTS.keys())),
```
Comment (sr229) [inline main.py]: This should reflect in API.md

Comment (Ovyerus) [reply main.py]: What do you mean?

Comment (sr229) [reply main.py]: not pointed out in API.md which are the supported font ranges

**Pull request merged**